### PR TITLE
Create Profile Image Upload/Display

### PR DESCRIPTION
**Pull request by JasperGroner**
Backend:

-Updated migration and model for users to include image url
-Added routes for uploading image in new profile and updating image on existing profile
-Added route for getting the image of the user who is signed in
-Modified review serializer to send back user image in addition to name
-Fixed bug in Add New Review where username, image, and votes were not being displayed because serializer was not being used

Frontend: 

-Added component section on form to include image upload
-Added link on topbar to change signed-in user image
-Added component (page) where user can change their image
-Added component displaying the logged in user's image
-Added the image of the user who reviewed an item to that review
-Edited styling to consolidate several redundant classes into one (centered-content) and make sure that was applied across the App (including to the change user image component)

### CODE REVIEW

File: client/src/components/SignedInUserImageTile.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React, { useState, useEffect } from "react";
+
+const SignedInUserImageTile = (props) => {
+	const [image, setImage] = useState({})
+
+	const getUserImage = async() => {
+		try {
+			const response = await fetch("/api/v1/users/image")
+			if (!response.ok) {
+        throw new Error(`${response.status} (${response.statusText})`)
```
Comment (MtJakhar) [inline client/src/components/SignedInUserImageTile.js:10]: I think there is an odd indentation here. Line 10

File: client/src/components/SignedInUserImageTile.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+			console.error(`Error in fetch: ${error.message}`)
+		}
+	}
+
+	useEffect(() => {
+		getUserImage()
+	}, [])
+
+	let imageDisplay
+	if(image) {
```
Comment (MtJakhar) [inline client/src/components/SignedInUserImageTile.js]: add space after if

Comment (nwalberts) [reply client/src/components/SignedInUserImageTile.js]: ill echo this

File: client/src/components/registration/RegistrationForm.js [hunk 3]
```diff
@@ -157,6 +166,10 @@ const RegistrationForm = () => {
 
           <input type="submit" className="button" value="Register" />
       </form>
+      <UploadUserImage 
+            setUserPayload={setUserPayload}
+            userPayload={userPayload}
+          />
```
Comment (nwalberts) [inline client/src/components/registration/RegistrationForm.js]: indentation ever so slightly off

File: client/src/components/uploads/ChangeUserImage.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+				headers: {
+					"Accept": "image/jpeg"
+				},
+				body: newUploadBody
+			})
+			if (!response.ok) {
+				throw new Error(`${response.status} (${response.statusText})`)
+			}
+			const body = await response.json()
+			console.log(body.updatedUser.image)
```
Comment (nwalberts) [inline client/src/components/uploads/ChangeUserImage.js]: uh oh! console.log

File: client/src/components/uploads/ChangeUserImage.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+						</section>
+					)}
+				</Dropzone>
+
+				<input className="button" type="submit" value="Change Profile Image" />
+			</form>
+
+			<div>
+				<img src={image} />
+			</div>
```
Comment (nwalberts) [inline client/src/components/uploads/ChangeUserImage.js]: this div isnt doing anything. delete it!

File: client/src/components/uploads/UploadUserImage.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	const addUpload = (event)=>{
+		event.preventDefault()
+		setUserPayload({
+			...userPayload,
+			image: newUploadFormData.image
+		})
+	}
+
+	return (
+		<div>
```
Comment (nwalberts) [inline client/src/components/uploads/UploadUserImage.js]: this div isnt doing anything, so replace with a React fragment

File: package.json [hunk 7]
```diff
@@ -18,5 +18,9 @@
   },
   "engines": {
     "node": "^16.18"
+  },
+  "dependencies": {
+    "@aws-sdk/client-s3": "^3.266.0",
+    "react-dropzone": "^14.2.3"
```
Comment (nwalberts) [inline package.json]: uh oh, dependencies should go into server/package.json or client/package.json

File: server/src/config.js [hunk 8]
```diff
@@ -2,9 +2,28 @@ import "./boot.js";
...
+}
+
+const config = { development, test, production }
+
+export default config[getNodeEnv()]
+
+
+
+
+
```
Comment (nwalberts) [inline server/src/config.js]: delete extra lines

File: server/src/db/migrations/20230208160119_addImageColumn.cjs [hunk 9]
```diff
@@ -0,0 +1,21 @@
+/**
+ * @typedef {import("knex")} Knex
+ */
+
+/**
+ * @param {Knex} knex
+ */
+exports.up = async (knex) => {
+  return knex.schema.table('users', table=> {
```
Comment (nwalberts) [inline server/src/db/migrations/20230208160119_addImageColumn.cjs]: add a space after `table`

**Review by nwalberts — CHANGES_REQUESTED** [review]

File: client/src/components/SignedInUserImageTile.js (hunk 2 above)
Comment (JasperGroner) [reply client/src/components/SignedInUserImageTile.js]: I thought I changed it and resubmitted! Brain fried.

**Pull request merged**